src/test/java/de/tum/in/www1/artemis/util/ProgrammingExerciseTestService.java

```diff
@@ -585,6 +585,7 @@ public void resumeProgrammingExerciseByTriggeringBuild_correctInitializationStat
         var participation = createStudentParticipationWithSubmission(exerciseMode);
         var participant = participation.getParticipant();
 
+        mockDelegate.mockTriggerParticipationBuild(participation);
```

Comment:
I personally would prefer to either move the second occurrence of `mockDelegate.mockTriggerParticipationBuild(participation);` a bit down (after the first call was executed) or add a comment that explains why the call must be mocked twice.
(The same applies to the other duplicated mocks)